### PR TITLE
re-enable h3 tests

### PR DESCRIPTION
**Pull request by olamy**

### CODE REVIEW

File: jetty-client/src/test/java/org/eclipse/jetty/client/Socks4ProxyTest.java [hunk 1]
```diff
@@ -318,7 +323,9 @@ public void testIdleTimeoutWhenSocksProxyDoesNotRespond() throws Exception
             // Accept the connection, but do not reply and don't close.
 
             ExecutionException x = assertThrows(ExecutionException.class, () -> listener.get(2 * idleTimeout, TimeUnit.MILLISECONDS));
-            assertThat(x.getCause(), instanceOf(TimeoutException.class));
+            Class<?> expectedException = "ci".equals(System.getProperty("env")) ? ConnectException.class : TimeoutException.class;
+            LOGGER.debug("exception during the request", x);
```
Comment (olamy) [inline jetty-client/src/test/java/org/eclipse/jetty/client/Socks4ProxyTest.java]: this is temporary for debugging I'm happy to remove it before merging.

File: jetty-client/src/test/java/org/eclipse/jetty/client/Socks4ProxyTest.java [hunk 2]
```diff
@@ -318,7 +323,9 @@ public void testIdleTimeoutWhenSocksProxyDoesNotRespond() throws Exception
             // Accept the connection, but do not reply and don't close.
 
             ExecutionException x = assertThrows(ExecutionException.class, () -> listener.get(2 * idleTimeout, TimeUnit.MILLISECONDS));
-            assertThat(x.getCause(), instanceOf(TimeoutException.class));
+            Class<?> expectedException = "ci".equals(System.getProperty("env")) ? ConnectException.class : TimeoutException.class;
```
Comment (olamy) [inline jetty-client/src/test/java/org/eclipse/jetty/client/Socks4ProxyTest.java]: I definitely reproduce it locally 100% of the time
```
[WARN] [stderr] Caused by: 
[WARN] [stderr] java.net.ConnectException: Connection refused
[WARN] [stderr] 	at java.base/sun.nio.ch.Net.pollConnect(Native Method)
[WARN] [stderr] 	at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
[WARN] [stderr] 	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:973)
[WARN] [stderr] 	at org.eclipse.jetty.io@10.0.13-SNAPSHOT/org.eclipse.jetty.io.SelectorManager.doFinishConnect(SelectorManager.java:341)
```

File: jetty-proxy/src/test/java/org/eclipse/jetty/proxy/ForwardProxyTLSServerTest.java [hunk 3]
```diff
@@ -910,7 +910,7 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
     @MethodSource("proxyTLS")
     public void testProxyLongProcessing(SslContextFactory.Server proxyTLS) throws Exception
     {
-        long timeout = 500;
+        long timeout = "ci".equals(System.getProperty("env")) ? 2000 : 500;
```
Comment (olamy) [inline jetty-proxy/src/test/java/org/eclipse/jetty/proxy/ForwardProxyTLSServerTest.java]: temporary as this means some flakyness so maybe even disable it if `"ci"`

File: jetty-proxy/src/test/java/org/eclipse/jetty/proxy/ForwardProxyTLSServerTest.java [hunk 4]
```diff
@@ -908,6 +907,7 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
 
     @ParameterizedTest
     @MethodSource("proxyTLS")
+    @DisabledIfSystemProperty(named = "env", matches = "ci")
```
Comment (olamy) [inline jetty-proxy/src/test/java/org/eclipse/jetty/proxy/ForwardProxyTLSServerTest.java]: I reproduce this locally for every run

File: jetty-client/src/test/java/org/eclipse/jetty/client/HttpClientTLSTest.java [hunk 5]
```diff
@@ -1377,8 +1377,6 @@ public void testForcedNonDomainSNI() throws Exception
     @EnabledForJreRange(max = JRE.JAVA_16, disabledReason = "Since Java 17, SNI host names can only have letter|digit|hyphen characters.")
     public void testForcedNonDomainSNIWithIPv6() throws Exception
     {
-        Assumptions.assumeTrue(Net.isIpv6InterfaceAvailable());
-
```
Comment (sbordet) [inline jetty-client/src/test/java/org/eclipse/jetty/client/HttpClientTLSTest.java]: This (and all other occurrences) should remain.
If IPv6 is available in the new Jenkins setup, then this assumption will pass and the test will run.

File: Jenkinsfile [hunk 6]
```diff
@@ -109,7 +109,7 @@ def mavenBuild(jdk, cmdline, mvnName) {
                "MAVEN_OPTS=-Xms2g -Xmx4g -Djava.awt.headless=true"]) {
         configFileProvider(
                 [configFile(fileId: 'oss-settings.xml', variable: 'GLOBAL_MVN_SETTINGS')]) {
-          sh "mvn --no-transfer-progress -s $GLOBAL_MVN_SETTINGS -Dmaven.repo.local=.repository -Pci -DexcludedGroups=\"external, large-disk-resource, stress, slow\" -V -B -e -Djetty.testtracker.log=true $cmdline"
+          sh "mvn -Dmaven.test.failure.ignore=true --no-transfer-progress -s $GLOBAL_MVN_SETTINGS -Dmaven.repo.local=.repository -Pci -DexcludedGroups=\"external, large-disk-resource, stress, slow\" -V -B -e -Djetty.testtracker.log=true $cmdline"
```
Comment (sbordet) [inline Jenkinsfile]: Why this change?

File: jetty-client/src/test/java/org/eclipse/jetty/client/Socks4ProxyTest.java (hunk 2 above)
Comment (sbordet) [reply jetty-client/src/test/java/org/eclipse/jetty/client/Socks4ProxyTest.java]: Don't do this. Maybe open an issue to be investigated, but if you make this change it will be forgotten.
Likely a Jenkins/Docker/K8s issue because I cannot reproduce it locally.

File: jetty-client/src/test/java/org/eclipse/jetty/client/Socks4ProxyTest.java (hunk 1 above)
Comment (sbordet) [reply jetty-client/src/test/java/org/eclipse/jetty/client/Socks4ProxyTest.java]: Please revert.

File: jetty-proxy/src/test/java/org/eclipse/jetty/proxy/ForwardProxyTLSServerTest.java (hunk 4 above)
Comment (sbordet) [reply jetty-proxy/src/test/java/org/eclipse/jetty/proxy/ForwardProxyTLSServerTest.java]: Don't disable it, let's try to find out why it fails.
Also this cannot be reproduced locally by me, so it's either Jenkins/Docker/K8s.

File: tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/HttpClientLoadTest.java [hunk 7]
```diff
@@ -82,8 +82,9 @@ public void init(Transport transport) throws IOException
     public void testIterative(Transport transport) throws Exception
     {
         // TODO: cannot run HTTP/3 (or UDP) in Jenkins.
-        if ("ci".equals(System.getProperty("env")))
-            Assumptions.assumeTrue(transport != Transport.H3);
+        // of course we can!
+//        if ("ci".equals(System.getProperty("env")))
+//            Assumptions.assumeTrue(transport != Transport.H3);
```
Comment (sbordet) [inline tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/HttpClientLoadTest.java]: Remove entirely comment and TODO.

File: tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/HttpClientLoadTest.java [hunk 8]
```diff
@@ -119,7 +120,8 @@ public void testIterative(Transport transport) throws Exception
     public void testConcurrent(Transport transport) throws Exception
     {
         // TODO: cannot run HTTP/3 (or UDP) in Jenkins.
-        Assumptions.assumeTrue(transport != Transport.H3);
+        // of course we can!
+        //Assumptions.assumeTrue(transport != Transport.H3);
```
Comment (sbordet) [inline tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/HttpClientLoadTest.java]: Remove comment and TODO.

**Review by sbordet — CHANGES_REQUESTED** [review]

File: jetty-client/src/test/java/org/eclipse/jetty/client/HttpClientTLSTest.java (hunk 5 above)
Comment (joakime) [reply jetty-client/src/test/java/org/eclipse/jetty/client/HttpClientTLSTest.java]: I agree with @sbordet - this assumption is also valid for users that build on machines that don't have IPv6 present/available.

File: Jenkinsfile (hunk 6 above)
Comment (joakime) [reply Jenkinsfile]: Yeah, why this change?

Comment (olamy) [reply Jenkinsfile]: I wanted to test to run ALL tests on a the new cluster. to know which one fail. without this the build is failing at the first test failure.
It's temporary and will not be merged.

File: jetty-client/src/test/java/org/eclipse/jetty/client/HttpClientTLSTest.java (hunk 5 above)
Comment (olamy) [reply jetty-client/src/test/java/org/eclipse/jetty/client/HttpClientTLSTest.java]: It's temporary and will be reverted as I wanted to be sure ipv6 tests were running! 
Maybe if someone could argue "blabla ipv6 tests are disabled" 🤣

File: jetty-client/src/test/java/org/eclipse/jetty/client/Socks4ProxyTest.java (hunk 2 above)
Comment (olamy) [reply jetty-client/src/test/java/org/eclipse/jetty/client/Socks4ProxyTest.java]: done

**Review by lorban — APPROVED** [review]

**Review by sbordet — APPROVED** [review]
LGTM, finger crossed.
